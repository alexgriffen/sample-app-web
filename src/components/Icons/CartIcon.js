import React, { Component } from 'react';
import { Badge, IconButton, withStyles } from '@material-ui/core';
import * as PropTypes from 'prop-types';
import cartIconPng from '../../assets/img/cart.png';
import cartIconSvg from '../../assets/svg/cart@3x.svg';
import { colors } from '../../utils/Colors';

const styles = () => ({
    root: {
      borderRadius: 0,
      '&:hover': {
        backgroundColor: colors.white,
      },
      padding: 0,
    },
    cartIcon: {
      background: `url(${ cartIconPng }) no-repeat center center`,
      backgroundImage: `url(${cartIconSvg})`,
      height: 40,
      width: 40,
    },
    badge: {
      backgroundColor: colors.slRed,
      top: '25%',
      right: 5,
    },
  }
);
const { func, number, object, string } = PropTypes;

class CartIcon extends Component {
  static propTypes = {
    cartContent: number.isRequired,
    classes: object.isRequired,
    fallBackClasses: string,
    onClick: func.isRequired,
    testID: string,
  };

  static defaultProps = {
    fallBackClasses: '',
    testID: 'cart-icon-id',
  };

  render() {
    const { cartContent, classes, fallBackClasses, onClick, testID } = this.props;

    return (
      <IconButton
        classes={ {
          // ` ${fallBackClasses}` is there to make it backwards compatible,
          // it doesn't not have an other function
          root: classes.root + ` ${ fallBackClasses }`,
        } }
        id={ testID }
        onClick={ onClick }
      >
        <Badge
          badgeContent={ cartContent }
          color="secondary"
          classes={ { badge: classes.badge } }
          id={ testID + '-content' }
        >
          <div className={ classes.cartIcon }/>
        </Badge>
      </IconButton>
    );
  }
}

export default withStyles(styles)(CartIcon);
