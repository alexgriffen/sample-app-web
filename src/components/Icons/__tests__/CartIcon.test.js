import React from 'react';
import { shallow } from 'enzyme/build';
import CartIcon from '../CartIcon';

let props;

describe('CartIcon', () => {
  beforeEach(() => {
    props = {
      cartContent: 0,
      onClick: jest.fn(),
    };
  });

  it('should render correctly with the required options', () => {
    const component = shallow(<CartIcon { ...props } />).dive();

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with full options', () => {
    const component = shallow(
      <CartIcon
        { ...props }
        cartContent={2}
        fallBackClasses="fallback-classes"
        testID="foo-test-bar-id"
      />
    ).dive();

    expect(component).toMatchSnapshot();
  });
});
