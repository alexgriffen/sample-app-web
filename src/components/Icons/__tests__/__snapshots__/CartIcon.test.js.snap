// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon should render correctly with full options 1`] = `
<WithStyles(IconButton)
  classes={
    Object {
      "root": "CartIcon-root-1 fallback-classes",
    }
  }
  id="foo-test-bar-id"
  onClick={[MockFunction]}
>
  <WithStyles(Badge)
    badgeContent={2}
    classes={
      Object {
        "badge": "CartIcon-badge-3",
      }
    }
    color="secondary"
    id="foo-test-bar-id-content"
  >
    <div
      className="CartIcon-cartIcon-2"
    />
  </WithStyles(Badge)>
</WithStyles(IconButton)>
`;

exports[`CartIcon should render correctly with the required options 1`] = `
<WithStyles(IconButton)
  classes={
    Object {
      "root": "CartIcon-root-1 ",
    }
  }
  id="cart-icon-id"
  onClick={[MockFunction]}
>
  <WithStyles(Badge)
    badgeContent={0}
    classes={
      Object {
        "badge": "CartIcon-badge-3",
      }
    }
    color="secondary"
    id="cart-icon-id-content"
  >
    <div
      className="CartIcon-cartIcon-2"
    />
  </WithStyles(Badge)>
</WithStyles(IconButton)>
`;
